discojs/src/dataset/dataset.ts

```diff
@@ -176,6 +216,23 @@ export class Dataset<T> implements AsyncIterable<T> {
     );
   }
 
+  /**
+   * Repeat the dataset `times` times
+   * @param times number of times to repeat the dataset, if undefined, the dataset is repeated indefinitely
+   * @returns a dataset repeated `times` times
+   */
+  repeat(times?: number): Dataset<T> {
+    return new Dataset(
+      async function* (this: Dataset<T>) {
+        let loop = 0;
+        do {
+          for await (const e of this) yield e;
```

Comment:
you can use the `yield*` meaning to yield everything from the given iterator.

```suggestion
          yield* this;
```